sample-app/src/components/SearchBar.tsx

```diff
@@ -33,16 +30,12 @@ export default function SearchBar({
   const answersActions = useAnswersActions();
   const query = useAnswersState(state => state.query.input);
   const isLoading = useAnswersState(state => state.searchStatus.isLoading);
-  /**
-   * Allow a query search to wait on the response to the autocomplete request right
-   * before the search execution in order to retrieve the search intents
-   */
-  const autocompletePromiseRef = useRef<Promise<AutocompleteResponse | undefined>>();
   const [autocompleteResponse, executeAutocomplete] = useSynchronizedRequest(() => {
     return isVertical
       ? answersActions.executeVerticalAutocomplete()
       : answersActions.executeUniversalAutocomplete();
   });
+  const [ executeQuery, autocompletePromiseRef ] = useNearMeSearch(answersActions, geolocationOptions);
```

Comment:
It feels strange that a method called `useNearMeSearch` is returning a ref to an auto-complete promise.